src/en/developer-layers.md

```diff
@@ -0,0 +1,153 @@
+
+# Why build from layers
+
+When creating a charm, you always have the option of doing it by
+hand writing.  That is, you can create each hook, implement the side of the
+interface you need for each relation your charm requires or provides, manage the
+dependencies, such as [charm-helpers][https://pythonhosted.org/charmhelpers/],
+that your charm uses, et cetera. However, what you really want to do is focus on
+*your* charm.  So, why not leverage the work of others and keep your charm code
+as minimal and tightly focused as possible?
+
+Enter the concept of building charms from layers.  Layers let you build on the
+work of other charmers, whether that work is in the form of other charms that
+you can extend and modify, interfaces that are already built for you and know
+how to communicate with a remote service and let you know when that service is
+ready and what it provides for you, or partial base layers that make managing
+dependencies much easier.  And it does this in a consistent, repeatable, and
+audit-able way.
+
+# What are layers?
+
+Layers are encapsulated charm code which lend themselves to being re-used across
+charms. They come in distinct flavors:
+
+- Base, or runtime layers Interface layers Charm layers
+
+Each of these has a distinct role, and it’s important to understand how a charm
+should be broken up into these types of layers. Generally, a charm will contain
+one base layer, one charm layer, and one or more interface layers, but it is
+possible that a charm might include more than one base layer, as well.
+
+## Base, or Runtime, Layers
+
+Base layers are layers that other charms can be built on. They contain things
+that are common to many different charms, and allow charms to reuse that
+commonality without having to reimplement it each time. Base layers typically
+are not sufficient on their own to be considered a charm; they likely can’t be
+built into a deployable charm, and if they can, they’re unlikely to do anything
+useful.
+
+The most basic example is just that,
+[layer-basic](http://github.com/juju-solutions/layer-basic). It provides nothing
+more than the minimum needed to effectively use layered charms:
+[charms.reactive](https://pythonhosted.org/charms.reactive/),
+[charmhelpers](https://pythonhosted.org/charmhelpers/), and the skeleton hook
+implementations that call into the reactive framework. However, the most useful
+base layers are actually a type of runtime layer. For example,
+[layer-apache-php](https://github.com/johnsca/apache-php) provides Apache2 and
+mod-php, as well as mechanisms for fetching and installing a PHP project within
+that runtime.
+
+Base layers can be written in any language, but must at a minimum provide the
+reactive framework that glues layers together, which is written in Python. This
+can be done trivially by building the base layer off of layer-basic.
+
+## Interface Layers
+
+Interface layers are perhaps the most confusing type of layer, and are
```

Comment:
I agree, and I think we will cover that in the in-depth w/ interface layers docs.